src/cpp/utils/Host.hpp

```diff
@@ -64,29 +66,24 @@ class Host
 
     Host()
     {
+        auto host_name = asio::ip::host_name();
```

Comment:
This function can throw an exception. There is an overload function that you pass the err_code as parameter.